volume/volume.go

```diff
@@ -174,18 +174,23 @@ func (m *MountPoint) Setup(mountLabel string, rootIDs idtools.IDPair, checkFun f
 	}()
 
 	if m.Volume != nil {
+		// This creates new dirs when a supath is mentioned along with the volume.
+		if err := idtools.MkdirAllAndChownNew(m.Source, 0755, rootIDs); err != nil {
+			return "", errors.Wrapf(err, "error creating SubPath of a volume %q", m.Source)
+		}
+
 		id := m.ID
 		if id == "" {
 			id = stringid.GenerateNonCryptoID()
 		}
-		path, err := m.Volume.Mount(id)
+		_, err := m.Volume.Mount(id)
```

Comment:
This needs to use the path returned from Mount as this is provided by the volume driver.
There is no guarantee that `mp.Source` is populated here.